coordinator/recovery/multi.go

```diff
@@ -52,6 +52,7 @@ func (r *MultiPartyRecovery) GenerateEncryptionKey(recoveryKeys map[string]strin
 	case len(recoveryKeys) <= 1:
 		r.log.Debug("Single recovery key received, generating single-party encryption key")
 		r.encryptionKey, err = generateRandomKey()
+		r.secretMap, r.SecretHashMap = nil, nil // wipe potentially previous multi-party recovery data
```

Comment:
we should rework this package such that the state of r does not change if a func returns an error